--- a/libhb/encx265.c
+++ b/libhb/encx265.c
@@ -463,10 +463,12 @@ int encx265Init(hb_work_object_t *w, hb_
     /* statsfile (but not 2-pass) */
     if (param->logLevel >= X265_LOG_DEBUG)
     {
-        if (param->csvfn == NULL)
+        if (param->csvfn[0] == '\0')
         {
-            pv->csvfn = hb_get_temporary_filename("x265.csv");
-            param->csvfn = strdup(pv->csvfn);
+            const char *tempFilename = hb_get_temporary_filename("x265.csv");
+            strncpy(param->csvfn, tempFilename, X265_MAX_STRING_SIZE - 1);
+            param->csvfn[X265_MAX_STRING_SIZE - 1] = '\0';
+            pv->csvfn = strdup(tempFilename);
         }
         else
         {
