diff -uNrp a/libhb/encx265.c b/libhb/encx265.c
--- a/libhb/encx265.c	2024-08-07 18:31:52.000000000 +0100
+++ b/libhb/encx265.c	2024-11-24 10:20:54.682279367 +0000
@@ -463,10 +463,10 @@ int encx265Init(hb_work_object_t *w, hb_
     /* statsfile (but not 2-pass) */
     if (param->logLevel >= X265_LOG_DEBUG)
     {
-        if (param->csvfn == NULL)
+        if (param->csvfn[0] == '\0')
         {
             pv->csvfn = hb_get_temporary_filename("x265.csv");
-            param->csvfn = strdup(pv->csvfn);
+            snprintf(param->csvfn, X265_MAX_STRING_SIZE, "%s", pv->csvfn);
         }
         else
         {
